crowbar_framework/app/models/service_object.rb

```diff
@@ -1412,6 +1423,15 @@ def apply_role(role, inst, in_queue)
       return [ 405, message ]
```

Comment:
Hrm. See the `return` here? If we return and don't restart the service, then we lose :-)